classes/ActionScheduler_ActionFactory.php

```diff
@@ -4,6 +4,37 @@
  * Class ActionScheduler_ActionFactory
  */
 class ActionScheduler_ActionFactory {
+
+	/**
+	 * @param string $status The action's status in the data store
+	 * @param string $hook The hook to trigger when this action runs
+	 * @param array $args Args to pass to callbacks when the hook is triggered
+	 * @param ActionScheduler_Schedule $schedule The action's schedule
+	 * @param string $group A group to put the action in
+	 *
+	 * @return ActionScheduler_StoredAction An instance of the stored action
+	 */
+	public function get_stored_action( $status, $hook, array $args = array(), ActionScheduler_Schedule $schedule = NULL, $group = '' ) {
+
+		switch ( $status ) {
+			case ActionScheduler_Store::STATUS_PENDING :
+				$action_class = 'ActionScheduler_Action';
+				break;
+			case ActionScheduler_Store::STATUS_CANCELED :
+				$action_class = 'ActionScheduler_CanceledAction';
+				break;
+			default :
+				$action_class = 'ActionScheduler_FinishedAction';
+				break;
+		}
+
+		$action_class = apply_filters( 'action_scheduler_stored_action_class', $action_class, $hook, $args, $schedule, $group );
+
+		$action = new $action_class( $hook, $args, $schedule, $group );
+
+		return apply_filters( 'action_scheduler_stored_action_instance', $action, $hook, $args, $schedule, $group );
```

Comment:
I've added one with SHA: 2a5dca27cc86e7892dd988e13732e51d6f448699. 

It feels quite redundant amidst the code though, because variable names, type declarations on the method and general understanding of what `apply_filters()` does, it feels like the docblock is just repeating the code.

Do you want to see these mostly for use with generated documentation? (i.e. docs outside the code)

Reply:
> Do you see these as valuable mainly for use with generated documentation? (i.e. docs outside the code) Or is it for something else?

It would certainly be useful for generated documentation, and I do hope we can make use of that. But I also think it's valuable to be consistent 🤷‍♂️